src/nupic/engine/Network.hpp

```diff
@@ -255,6 +256,14 @@ namespace nupic
     getRegions() const;
 
     /**
+     * Get all links between regions
+     *
+     * @returns A Collection of Link objects in the network
+     */
+    const Collection<Link *>
+    getLinks() const;
```

Comment:
I'm guessing we should just remove the `const`. The reason it's used for `getRegions` is because it's returning a const reference `const Collection<Region*>&`. I don't see much use in returning a const value. And we're probably fine with the caller modifying the link, so it doesn't need to be a `const Link *`.